pallets/manta-pay/src/lib.rs

```diff
@@ -1402,3 +1406,278 @@ where
+        _sinks: I,
+    ) -> Result<Vec<Self::ValidSinkAccount>, InvalidSinkAccount<config::Config, Self::AccountId>>
+    where
+        I: Iterator<Item = (Self::AccountId, config::AssetValue)>,
+    {
+        // No Sinks for SBT
+        Ok(Vec::new())
+    }
+
+    #[inline]
+    fn is_valid(
+        &self,
+        posting_key: TransferPostingKeyRef<config::Config, Self>,
+    ) -> Result<(Self::ValidProof, Self::Event), TransferLedgerError<T>> {
+        let transfer_shape = TransferShape::from_posting_key_ref(&posting_key);
+        let (mut verifying_context, event) =
+            match transfer_shape.ok_or(TransferLedgerError::InvalidTransferShape)? {
+                TransferShape::ToPrivate => {
+                    if let Some(asset_id) = posting_key.asset_id.or(None) {
+                        let asset_id =
+                            fp_encode(asset_id).map_err(TransferLedgerError::FpEncodeError)?;
+                        (
+                            manta_parameters::pay::verifying::ToPrivate::get()
+                                .ok_or(TransferLedgerError::ChecksumError)?,
+                            PreprocessedEvent::<T>::ToPrivate {
+                                asset: Asset::new(
+                                    asset_id,
+                                    asset_value_encode(posting_key.sources[0].1),
+                                ),
+                                source: posting_key.sources[0].0.clone(),
+                            },
+                        )
+                    } else {
+                        return Err(TransferLedgerError::UnknownAsset);
+                    }
+                }
+                TransferShape::PrivateTransfer => {
+                    return Err(TransferLedgerError::SenderLedgerError(
+                        SenderLedgerError::NoSenderLedger,
+                    ))
+                }
+                TransferShape::ToPublic => {
+                    return Err(TransferLedgerError::SenderLedgerError(
+                        SenderLedgerError::NoSenderLedger,
+                    ))
+                }
+            };
+        let verification = posting_key
+            .has_valid_proof(
+                &config::VerifyingContext::decode(&mut verifying_context)
+                    .map_err(TransferLedgerError::VerifyingContextDecodeError)?,
+            )
+            .map_err(TransferLedgerError::ProofSystemError)?;
+        if verification {
+            Ok((Wrap(()), event))
+        } else {
+            Err(TransferLedgerError::InvalidProof)
+        }
+    }
+
+    #[inline]
+    fn update_public_balances(
+        &mut self,
+        _super_key: &TransferLedgerSuperPostingKey<config::Config, Self>,
+        _asset_id: config::AssetId,
+        _sources: Vec<SourcePostingKey<config::Config, Self>>,
+        _sinks: Vec<SinkPostingKey<config::Config, Self>>,
+        _proof: Self::ValidProof,
+    ) -> Result<(), TransferLedgerError<T>> {
+        Ok(())
```

Comment:
Same here, confused by the function name.

Reply:
same it is a trait from `manta-rs`